File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/node/DecryptErrorMsgBlock.java
/*
 * © 2016-2019 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.retrofit.response.model.node;

import android.os.Parcel;

import com.google.gson.annotations.Expose;

/**
 * @author Denis Bondarenko
 * Date: 3/26/19
 * Time: 3:02 PM
 * E-mail: DenBond7@gmail.com
 */
public class DecryptErrorMsgBlock extends MsgBlock {
  public static final Creator<DecryptErrorMsgBlock> CREATOR = new Creator<DecryptErrorMsgBlock>() {
    @Override
    public DecryptErrorMsgBlock createFromParcel(Parcel source) {
      return new DecryptErrorMsgBlock(source, Type.DECRYPT_ERROR);
    }

    @Override
    public DecryptErrorMsgBlock[] newArray(int size) {
      return new DecryptErrorMsgBlock[size];
    }
  };

  @Expose
  private DecryptError decryptErr;

  public DecryptErrorMsgBlock() {
  }

  protected DecryptErrorMsgBlock(Parcel in, Type type) {
    super(in, type);
    this.decryptErr = in.readParcelable(DecryptError.class.getClassLoader());
  }

  @Override
  public int describeContents() {
    return 0;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    super.writeToParcel(dest, flags);
    dest.writeParcelable(this.decryptErr, flags);
  }

  public DecryptError getError() {
    return decryptErr;
  }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/node/DecryptErrorMsgBlock.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.retrofit.response.model.node

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * @author Denis Bondarenko
 * Date: 3/26/19
 * Time: 3:02 PM
 * E-mail: DenBond7@gmail.com
 */
data class DecryptErrorMsgBlock(
  @Expose override val content: String?,
  @Expose override val complete: Boolean,
  @SerializedName("decryptErr") @Expose val error: DecryptError?,
  /*// TODO: remove above and change name of below when finally dropping Node
  @SerializedName("kotlinDecryptErr") @Expose val kotlinError: PgpMsg.DecryptionError? = null*/
) : MsgBlock {

  @Expose
  override val type: MsgBlock.Type = MsgBlock.Type.DECRYPT_ERROR

  constructor(source: Parcel) : this(
    source.readString(),
    1 == source.readInt(),
    source.readParcelable<DecryptError>(DecryptError::class.java.classLoader)/*,
    source.readParcelable<PgpMsg.DecryptionError>(PgpMsg.DecryptionError::class.java.classLoader)*/
  )

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) =
    with(dest) {
      writeParcelable(type, flags)
      writeString(content)
      writeInt(if (complete) 1 else 0)
      writeParcelable(error, flags)/*
      writeParcelable(kotlinError, flags)*/
    }

  companion object {
    @JvmField
    val CREATOR: Parcelable.Creator<DecryptErrorMsgBlock> =
      object : Parcelable.Creator<DecryptErrorMsgBlock> {
        override fun createFromParcel(source: Parcel): DecryptErrorMsgBlock {
          source.readParcelable<MsgBlock.Type>(MsgBlock.Type::class.java.classLoader)
          return DecryptErrorMsgBlock(source)
        }

        override fun newArray(size: Int): Array<DecryptErrorMsgBlock?> = arrayOfNulls(size)
      }
  }
}
